osu.Game/Rulesets/Mods/ModTimeRamp.cs

```diff
@@ -29,20 +29,26 @@ public abstract class ModTimeRamp<T> : ModTimeRamp, IUpdatableByPlayfield, IAppl
 
         private IAdjustableClock clock;
 
-        private IHasPitchAdjust pitchAdjust;
-
         /// <summary>
         /// The point in the beatmap at which the final ramping rate should be reached.
         /// </summary>
         private const double final_rate_progress = 0.75f;
 
+        /// <summary>
+        /// The adjustment applied on entering this mod's application method.
+        /// </summary>
+        private double baseAdjust;
+
         public virtual void ApplyToClock(IAdjustableClock clock)
         {
             this.clock = clock;
-            pitchAdjust = (IHasPitchAdjust)clock;
 
-            // for preview purposes
-            pitchAdjust.PitchAdjust = 1.0 + FinalRateAdjustment;
+            // we capture the adjustment applied before entering our application method.
+            // this will cover external changes, which should re-fire this method.
+            baseAdjust = (clock as IHasPitchAdjust)?.PitchAdjust ?? clock.Rate;
+
+            // for preview purposes. during gameplay, Update will overwrite this setting.
+            applyAdjustment(1);
```

Comment:
This is seemingly doing nothing?

Reply:
It affects the rate at song select.

Reply:
But it's 1

Reply:
yeah, applied the rate at final adjustment rate (see usage of parameter). this is basically what it did previous to this PR, just sharing logic.

Reply:
Read the code in `applyAdjustment()` again, you changed its behaviour, the value provided in is the intended adjustment, not a value representing a scale of the final adjustment.